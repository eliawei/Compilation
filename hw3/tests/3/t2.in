// check +/- operands for set
void main() {
    	int x = 5;
    	byte y = 10b;
	bool z;

  set[0..255] myset1;
	// - operand
    	set[0..255] myset2 = myset1 - x;
	myset2 = x - myset1;
	myset2 = myset1 - y;
	myset2 = y - myset1;
	z = x in myset2;

	// + operand
	myset2 = myset1 + x;
	myset2 = x + myset1;
	myset2 = myset1 + y;
	myset2 = y + myset1;

  z = x in myset2;
}
