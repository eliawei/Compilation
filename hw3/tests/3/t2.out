---end scope---
x INT 0
y BYTE 1
z BOOL 2
myset1 SET 3
myset2 SET 4
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
main ()->VOID 0
